### PR TITLE
docs: document about distribute tracing

### PR DESCRIPTION
**Pull request by Taylor-lagrange**
## What's Changed in this PR

Add document about distribute tracing

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [x] This change requires follow-up update in localized docs.

### CODE REVIEW

**Review by zhaoyingnan01 — APPROVED** [review]

File: docs/v0.4/en/developer-guide/how-to/how-to-trace-greptimedb.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+GreptimeDB uses Rust's [tracing](https://docs.rs/tracing/latest/tracing/) framework for code instrument. For the specific details and usage of tracing, please refer to the official documentation of tracing.
+
+By transparently transmitting `trace_id` and other information on the entire distributed system, we can record the function call chain of the entire distributed link, know the time of each tracked function take and other related information, so as to monitor the entire system.
+
+## Define tracing context in rpc
+
+Because the tracing framework does not natively support distributed tracing, we need to manually pass information such as `trace_id` in the rpc message to correctly identify the function calling relationship. We use standards based on [w3c](https://www.w3.org/TR/trace-context/#traceparent-header-field-values) to encode relevant information into `tracing_context` and attach the message to the rpc header. Mainly defined in:
+
+- frontend interacts with datanode: `tracing_context` is defined in [`RegionRequestHeader`](https://github.com/GreptimeTeam/greptime-proto/blob/main/proto/greptime/v1/region/server.proto)
```
Comment (nicecui) [inline docs/v0.4/en/developer-guide/how-to/how-to-trace-greptimedb.md:11]:
```suggestion
- `frontend` interacts with `datanode`: `tracing_context` is defined in [`RegionRequestHeader`](https://github.com/GreptimeTeam/greptime-proto/blob/main/proto/greptime/v1/region/server.proto)
```

File: docs/v0.4/en/developer-guide/how-to/how-to-trace-greptimedb.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+- frontend interacts with metasrv: `tracing_context` is defined in [`RequestHeader`](https://github.com/GreptimeTeam/greptime-proto/blob/main/proto/greptime/v1/meta/common.proto)
```
Comment (nicecui) [inline docs/v0.4/en/developer-guide/how-to/how-to-trace-greptimedb.md:12]:
```suggestion
- `frontend` interacts with `metasrv`: `tracing_context` is defined in [`RequestHeader`](https://github.com/GreptimeTeam/greptime-proto/blob/main/proto/greptime/v1/meta/common.proto)
```

File: docs/v0.4/en/developer-guide/how-to/how-to-trace-greptimedb.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+- Client interacts with frontend: `tracing_context` is defined in [`RequestHeader`](https://github.com/GreptimeTeam/greptime-proto/blob/main/proto/greptime/v1/common.proto)
```
Comment (nicecui) [inline docs/v0.4/en/developer-guide/how-to/how-to-trace-greptimedb.md:13]:
```suggestion
- Client interacts with `frontend`: `tracing_context` is defined in [`RequestHeader`](https://github.com/GreptimeTeam/greptime-proto/blob/main/proto/greptime/v1/common.proto)
```

File: docs/v0.4/en/user-guide/operations/configuration.md [hunk 4]
```diff
@@ -406,6 +406,27 @@ use_memory_store = false
 | selector         | String  | Load balance strategy to choose datanode when creating new tables, see [Selector](/developer-guide/metasrv/selector.md)                 |
 | use_memory_store | Boolean | Only used for testing when you don't have an etcd cluster, store data in memory, `false` by default.                                    |
 
+### Logging options
+
+frontend, metasrv, datanode and standalone can all configure log and tracing related parameters in the `[logging]` section:
```
Comment (nicecui) [inline docs/v0.4/en/user-guide/operations/configuration.md:411]:
```suggestion
`frontend`, `metasrv`, `datanode` and standalone can all configure log and tracing related parameters in the `[logging]` section:
```

File: docs/v0.4/en/user-guide/operations/configuration.md [hunk 5]
```diff
@@ -406,6 +406,27 @@ use_memory_store = false
...
+[logging]
+dir = "/tmp/greptimedb/logs"
+level = "info"
+enable_otlp_tracing = false
+otlp_endpoint = "localhost:4317"
+tracing_sample_ratio = 1.0
+```
+
+- `dir`: log output directory.
+- `level`: log output log level, the default level is `info`.
```
Comment (nicecui) [inline docs/v0.4/en/user-guide/operations/configuration.md:423]: any other levels?

**Review by nicecui — COMMENTED** [review]
👍  Almost done, just a little change left:

`rpc` -> `RPC`

File: docs/v0.4/en/user-guide/operations/tracing.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
+# Tracing
+
+GreptimeDB supports distributed tracing. GreptimeDB exports all collected spans using the grpc-based OTLP protocol. Users can use [Jaeger](https://www.jaegertracing.io/), [Tempo](https://grafana.com/oss/tempo/) and other OTLP protocol backends that support grpc to collect the span instrument by GreptimeDB.
```
Comment (nicecui) [inline docs/v0.4/en/user-guide/operations/tracing.md:3]:
```suggestion
GreptimeDB supports distributed tracing. GreptimeDB exports all collected spans using the gRPC-based OTLP protocol. Users can use [Jaeger](https://www.jaegertracing.io/), [Tempo](https://grafana.com/oss/tempo/) and other OTLP protocol backends that support grpc to collect the span instrument by GreptimeDB.
```

**Review by nicecui — COMMENTED** [review]
gprc -> `gRPC`

**Review by nicecui — APPROVED** [review]

**Pull request merged**